src/Storage/Storage.Management/File/NewAzureStorageShare.cs

```diff
@@ -0,0 +1,125 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Management.Storage.Models;
+using Microsoft.Azure.Management.Storage;
+using Microsoft.Azure.Management.Storage.Models;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Management.Storage
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMStoragePrefix + StorageShareNounStr, DefaultParameterSetName = AccountNameParameterSet, SupportsShouldProcess = true), OutputType(typeof(PSShare))]
+    public class NewAzureStorageShareCommand : StorageFileBaseCmdlet
+    {
+        /// <summary>
+        /// AccountName Parameter Set
+        /// </summary>
+        private const string AccountNameParameterSet = "AccountName";
+
+        /// <summary>
+        /// Account object parameter set 
+        /// </summary>
+        private const string AccountObjectParameterSet = "AccountObject";
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.",
+            ParameterSetName = AccountNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Storage Account Name.",
+            ParameterSetName = AccountNameParameterSet)]
+        [Alias(AccountNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string StorageAccountName { get; set; }
+
+        [Parameter(Mandatory = true,
+            HelpMessage = "Storage account object",
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = AccountObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSStorageAccount StorageAccount { get; set; }
+
+        [Alias("N", "ShareName")]
+        [Parameter(Mandatory = true, 
+            HelpMessage = "Azure File share name",
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "Share Quota in Gibibyte.")]
```

Comment:
GigaByte.  Was this the agreed on parameter name?  It should likely be QuotaGB, or QuotaInGB

Reply:
The parameter name is raised by feature team PM. I have discussed this parameter name with Feature PM, and this should be GiB (Gibibyte), which is different from GB (Gigabyte), see [link](https://en.wikipedia.org/wiki/Gibibyte)